src/main/java/de/retest/recheck/configuration/ProjectRootFinder.java

```diff
@@ -0,0 +1,12 @@
+package de.retest.recheck.configuration;
+
+import java.nio.file.Path;
+import java.util.Optional;
+
+public interface ProjectRootFinder {
+
+	Optional<Path> findProjectRoot();
```

Comment:
I think it would helpful to describe both methods with some Javadoc? Where does `findProjectRoot()` start the search? What happens with `basePath`?